client/src/components/game/gui/panels/map/MapPanel.jsx

```diff
@@ -1,10 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
+import {
+    MapContainer, TileLayer, Marker, useMapEvents, Tooltip,
+} from "react-leaflet";
+import Leaflet, { CRS } from "leaflet";
 import PanelTemplate from "../panel_template/PanelTemplate";
 import mapIcon from "../../../../../assets/images/gui/hud/map-icon.png";
+import playerIcon from "../../../../../assets/images/gui/panels/map/player-marker.png";
+import destinationIcon from "../../../../../assets/images/gui/panels/map/destination-marker.png";
+import "leaflet/dist/leaflet.css";
 import "./MapPanel.scss";
+import { ApplicationState, PlayerState } from "../../../../../shared/state/States";
+
+const markerScale = 4;
+
+const playerMarker = Leaflet.icon({
+    iconUrl: playerIcon,
+    iconSize: [12 * markerScale, 14 * markerScale],
+    iconAnchor: [24, 48],
+});
+
+const locationMarker = Leaflet.icon({
+    iconUrl: destinationIcon,
+    iconSize: [12 * markerScale, 14 * markerScale],
+    iconAnchor: [24, 48],
+});
+
+function LocationMarker() {
+    const [position, setPosition] = useState(null);
+
+    useMapEvents({
```

Comment:
The + 0.5 trick to make the marker icon be properly centred on the target tile will be needed for this location marker too. 
![image](https://user-images.githubusercontent.com/9785268/111054351-e2679500-8463-11eb-9035-a71ef153e7bf.png)

![image](https://user-images.githubusercontent.com/9785268/111054386-4be7a380-8464-11eb-978a-c1ad81599d5c.png)